docs/source/serving/sky-serve.rst

```diff
@@ -521,3 +521,64 @@ The :code:`resources` field has the same spec as a normal SkyPilot job; see `her
   These settings will not take effect if you have an existing controller (either
   stopped or live).  For them to take effect, tear down the existing controller
   first, which requires all services to be terminated.
+
+Speedup Weights Loading in Large Model Serving
+----------------------------------------------
+
+When serving large models, the weights of the model are loaded from the cloud storage / public internet to the VMs. This process can take a long time, especially for large models. To speed up the weights loading, you can use the following methods:
+
+Use Machine image
+~~~~~~~~~~~~~~~~~
+
+You can use a machine image that has the weights preloaded. This can be done by creating a VM, loading the weights, and then creating a machine image from the VM. Then, you can use the machine image to launch the VMs for serving:
+
+.. code-block:: yaml
+  :emphasize-lines: 7-8
+
+    service:
+      replicas: 2
+      readiness_probe: /v1/models
+
+    resources:
+      accelerators: A100:1
+      cloud: gcp
+      image_id: projects/my-project/global/machineImages/image-with-model-weights
+
+Notice that the :code:`cloud` field must be specified when :code:`image_id` is used. You can use :code:`any_of` in :code:`resources` to make it support multiple clouds:
+
+.. code-block:: yaml
+  :emphasize-lines: 7-13
+
+    service:
+      replicas: 2
+      readiness_probe: /v1/models
+
+    resources:
+      accelerators: A100:1
+      any_of:
+        - cloud: gcp
+          image_id: projects/my-project/global/machineImages/image-with-model-weights
+        - cloud: aws
+          image_id:
+            us-east-1: ami-0729d913a335efca7
+            us-west-2: ami-050814f384259894c
+
+Notice that AWS needs a per-region image since its image is only valid in a single region. This requires to configure a lot of images on the cloud console, but it can significantly speed up the weights loading.
+
+Use Docker Container as Runtime Environment
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You can also :ref:`use docker containers as runtime environment <docker-containers-as-runtime-environments>`. This can be done by creating a Docker container that has the weights preloaded, and then using the container to launch the VMs for serving:
+
+.. code-block:: yaml
+  :emphasize-lines: 7
+
+    service:
+      replicas: 2
+      readiness_probe: /v1/models
+
+    resources:
+      accelerators: A100:1
+      image_id: docker:docker-image-with-model-weights
+
```

Comment:
We should still have `run` section here, as we don't run the ENTRYPOINT of the docker image?

Reply:
Maybe we can just recommend `docker run` in the `run` section directly?

Reply:
Sorry, I mean instead of specifying the docker image in `image_id:`, we just ask the user to specify `docker run custom-image` in the `run` section of our yaml.

Reply:
This was originally posted by zongheng [here](https://skypilot-org.slack.com/archives/C03J2KQQZSS/p1706847782816049?thread_ts=1706841328.945889&cid=C03J2KQQZSS) and actually I'm a little bit confused after a second look: will it accelerate the process? We still need to download from the docker registry and what is the difference compared to downloading from, say, hugging face..? cc @concretevitamin for a look

Reply:
Oh Zhanghao just explain that to me in the following comments. nvm :)